### PR TITLE
feat: Follow up to ENT-4779 PR to handle in loop exceptions

### PR DESCRIPTION
**Pull request by binodpant**
follow up to https://github.com/edx/edx-enterprise/pull/1312 per Brian's feedback

If one email send fails for any reason, avoid failure and log exception, includes a test

ENT-4779

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [x] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [x] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

**Comment by binodpant** [comment]
will do version and changelog per merge

File: enterprise/tasks.py [hunk 1]
```diff
@@ -40,14 +40,23 @@ def send_enterprise_email_notification(
...
+            try:
+                send_email_notification_message(
+                    item['user'],
+                    item['enrolled_in'],
+                    item['dashboard_url'],
+                    enterprise_customer_uuid,
+                    email_connection=email_conn,
+                    admin_enrollment=admin_enrollment,
+                )
+            except Exception as exc:  # pylint: disable=broad-except
```
Comment (binodpant) [inline enterprise/tasks.py:53]: open to thoughts but I don't see much of a way around catching all exceptions, I have no idea what specific exceptions to catch here

File: tests/test_enterprise/test_tasks.py [hunk 2]
```diff
@@ -106,6 +107,47 @@ def test_create_enrollment_task_no_create_duplicates(self, catalog_api_client_mo
...
+
+        # check exception is logged at ERROR level twice, in this case
+        # but causes no failures otherwise
+        with self.assertLogs('enterprise.tasks', level='ERROR') as log_cm:
+            send_enterprise_email_notification(
+                enterprise_customer.uuid,
+                True,
+                email_items,
+            )
+            self.assertEqual(2, len(log_cm.records))
```
Comment (binodpant) [inline tests/test_enterprise/test_tasks.py:148]: just checking in this case that two logger records got created

**Review by macdiesel — APPROVED** [review]
Cool, thank you!

File: enterprise/tasks.py [hunk 3]
```diff
@@ -52,9 +56,10 @@ def send_enterprise_email_notification(
                 )
             except Exception as exc:  # pylint: disable=broad-except
                 LOGGER.exception(
-                    f"Failed notifying user: "
-                    f"enterprise_customer_uuid: {enterprise_customer_uuid}"
-                    f"of enterprise enrollment in course {course_name}",
+                    "Failed notifying user: {}"
```
Comment (binodpant) [inline enterprise/tasks.py:59]: do not want to print email etc pii, so just printing username if available

**Pull request merged**